src/PrivateDns/PrivateDns/Zones/SetAzurePrivateDnsZone.cs

```diff
@@ -0,0 +1,104 @@
+﻿// ------------------------------------------------------------------------------------------------
+// <copyright file="SetAzurePrivateDnsZone.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+// ------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.PrivateDns.Zones
+{
+    using System.Collections;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.PrivateDns.Models;
+    using Microsoft.Azure.Commands.PrivateDns.Utilities;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using ProjectResources = Microsoft.Azure.Commands.PrivateDns.Properties.Resources;
+
+    /// <summary>
+    /// Updates an existing zone.
+    /// </summary>
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "PrivateDnsZone", SupportsShouldProcess = true, ConfirmImpact = ConfirmImpact.Medium), OutputType(typeof(PrivateDnsZone))]
+    public class SetAzurePrivateDnsZone : PrivateDnsBaseCmdlet
+    {
+        private const string FieldsParameterSetName = "Fields";
+        private const string ObjectParameterSetName = "Object";
+        private const string ResourceParameterSetName = "ResourceId";
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The full name of the zone (without a terminating dot).", ParameterSetName = FieldsParameterSetName)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource group in which the zone exists.", ParameterSetName = FieldsParameterSetName)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "A hash table which represents resource tags.", ParameterSetName = FieldsParameterSetName)]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "A hash table which represents resource tags.", ParameterSetName = ResourceParameterSetName)]
+        public Hashtable Tags { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The zone object to set.", ParameterSetName = ObjectParameterSetName)]
+        [ValidateNotNullOrEmpty]
+        public PrivateDnsZone Zone { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Do not use the ETag field of the RecordSet parameter for optimistic concurrency checks.", ParameterSetName = ObjectParameterSetName)]
+        [Parameter(Mandatory = false, HelpMessage = "Do not use the ETag field of the RecordSet parameter for optimistic concurrency checks.", ParameterSetName = ResourceParameterSetName)]
+        public SwitchParameter Overwrite { get; set; }
```

Comment:
This param does not apply to Fields param set?

Reply:
no actually it doesn't apply to resourceid param set either. will remove it